### ISSUE TITLE
Training function

### ISSUE THREAD

**Issue by ManuelPopp** [ManuelPopp/MUSTAFAR#2]
Write adaptable training loop which allows use of callbacks and returns either the last or the best (on validation set) model, as well as training history in a pandas dataframe (for the sake of easy plotting and export).
